backend/src/server_host.py

```diff
@@ -383,18 +371,20 @@ async def install_deps(dependencies: list[api.Dependency]):
                 to_install.append(dep)
 
     try:
-        if len(to_install) > 0:
-            await worker.stop()
+        deps_to_install = filter_necessary_to_install(deps_to_dep_info(to_install))
 
-            await install_deps(to_install)
-            flags = []
-            if config.error_on_failed_node:
-                flags.append("--error-on-failed-node")
-
-            if config.close_after_start:
-                flags.append("--close-after-start")
+        restart_flags: list[str] = []
+        if config.error_on_failed_node:
+            restart_flags.append("--error-on-failed-node")
+        if config.close_after_start:
+            restart_flags.append("--close-after-start")
 
-            await worker.start(flags)
+        if len(restart_flags) > 0 or len(deps_to_install) > 0:
```

Comment:
why are the flags that we've passed in a condition that would make this restart? we only need to restart if we installed any deps. the flag stuff is just so that when we restart, we restart with the flags we need. we already have those flags under a normal start

Reply:
> we already have those flags under a normal start

Nope. The first start is always with no flags.

Reply:
Then that's a bug isn't it?

Reply:
Actually, maybe that was intentional -- we probably don't want to do the error on failed node or close after start flags until the deps are installed properly, so that does make sense.

never mind then